### PR TITLE
Updated preprocessing and data evaluation

### LINKED ISSUE #25: Clarify image classification details
Need to specify input resolution as well. Also specify separate Top1 and Top5 scores. (Edit: Maybe, needs to be one quality target.)

Per User guide comment by Ableiweiss.

### LINKED ISSUE #30: Formalize input preprocessing
In our experience, input preprocessing may significantly affect accuracy scores.

For example, when evaluating the [Intel Caffe artifact](https://github.com/ctuning/ck-request-asplos18-caffe-intel) submitted to [ReQuEST@ASPLOS'18](http://cknowledge.org/request-cfp-asplos2018.html), we measured the following Top-1/Top-5 ImageNet [scores](https://github.com/ctuning/ck-request-asplos18-caffe-intel/issues/7#issuecomment-367319271) for [ResNet50](https://arxiv.org/abs/1512.03385) in single-precision floating-point:

Resolution | Top-1 | Top-5
------|--------------|-------------
224 | 0.707186 | 0.898327
256 | 0.729173 | 0.911852
320 | 0.731754 | 0.912832
320' | 0.736136 | 0.914713

This table shows that while ResNet50 [operates](https://github.com/KaimingHe/deep-residual-networks/blob/master/prototxt/ResNet-50-deploy.prototxt) on _224x224_ input images, rescaling to a higher resolution and then cropping to _224x224_ can boost the Top-1 accuracy score by up to 3%! 

Furthermore, the accuracy may significantly depend on the mean values used. In the above table, the row indexed _320_ was obtained with the mean calculated over the _validation_ set, while the row indexed _320'_ with the mean calculated over the _training_ set.

It is clear that this or similar preprocessing techniques can in principle be used to hide significant accuracy degradation from aggressive quantization or pruning techniques. Therefore, their use should be formalized.

Unspecified preprocessing could also be the reason why we [struggled](https://github.com/ARM-software/ComputeLibrary/issues/431) to validate the [claimed MobileNets-v1 accuracy](https://github.com/tensorflow/models/blob/1630da3434974e9ad5a0b6d887ac716a97ce03d3/research/slim/nets/mobilenet_v1.md) against [our own implementations](https://github.com/dividiti/ck-request-asplos18-mobilenets-armcl-opencl) using TensorFlow and the Arm Compute Library, with **the same pretrained weights**.

This is relevant not just for inference but for training too. Collecting data about the costs and gains of preprocessing, especially for techniques like [progressive resizing](http://www.fast.ai/2018/04/30/dawnbench-fastai/), could also help autotune the training process.

**NB:** Originally opened as https://github.com/mlperf/reference/issues/48

### LINKED ISSUE #31: Clarify definition of app to prevent cache tricks
Metric is defined as end to end execution of application. Is there a clever way to preload outside of this definition?
==
If the application is interpreted to be the python script, then that would exclude things like scp-ing data from GCS/S3 and/or running a block-cache pre-warmer to load the dataset into the Linux block cache (and/or taking the second training run because the first one was cancelled mid-way through / etc). Maybe it makes sense to provide guidance on the interpretation of application? What do you think?

### LINKED ISSUE #58: Padding
Seeking clarity on when and where you can pad.

### LINKED ISSUE #61: Clarification of Dataset location at clock start
What are the requirements of where the data lives when the clock starts?

I believe we have already (implicitly) clarified that data cannot reside in RAM when the clock starts (based on the decision requiring "cold starts" for all caches). 

- Can the dataset live in a distributed file system (e.g. cloud use cases)?
- Can the dataset live on the local disk?
- Are the limitations for a valid local disks (spindle, SSD, raid, something very fancy)? 
- Can the data already be unzipped (some reference implementations time the unzipping of the data, though I believe the rules state such a step does not need to be timed)?

### LINKED ISSUE #67: Max Length Truncation
Can you truncate long examples in translation datasets? This may enable certain hardware. On face value, we would expect this to risk reducing quality and not yield an advantage (but we lack data).

### LINKED ISSUE #75: Shard size can change randomization order
Submitters WG seems to accept as standard practice, just making explicit

### PR DESCRIPTION
**Pull request by petermattson**
Fixes #30	Preprocessing that can be used for multiple runs may be done in advance (e.g. does not include random transformations). 
	Preprocessing done in advance may include 
	Only pre-processing that must be done for each run will be timed in v0.5.
Fixes #61	Data can start on any durable storage system such as local disks and cloud storage systems. This explicitly excludes RAM.
Fixes #31	You must flush the cache or restart the system prior to benchmarking
Fixes #25 Images must have the same size as in the reference implementation. 
Fixes #58	Mathematically equivalent padding of images is allowed.
Fixes #67	For benchmarks with sequence inputs, you may choose a length N and either truncate all examples to length N or throw out all examples which exceed length N. This must be done uniformly for all examples. This may only be done on the training set and not the evaluation set.
Fixes #75	Shard size can change data randomization order

### CODE REVIEW

File: training_rules.adoc [hunk 1]
```diff
@@ -165,20 +165,32 @@ The qualified name “MLPerf Open” must be used when referring to an Open Divi
...
+=== Preprocessing During the Run
+Only preprocessing that must be done for each run (e.g. random transformations) must be timed.
 
 CLOSED: The same preprocessing steps as the reference implementation must be used. 
 
 OPEN: Any preprocessing steps are allowed. However, each datum must be preprocessed individually in a manner that is not influenced by any other data.
 
+=== Data Representation
+
+CLOSED: Images must have the same size as in the reference implementation. Mathematically equivalent padding of images is allowed.
```
Comment (nvpaulius) [inline training_rules.adoc:183]: Are we referring to the image sizes after reformatting (as part of preprocessing before the timed run) or images as being submitted to the network during training? I think it's the former. I wonder if we need to give more detail for what we mean by 'mathematially equivalent padding' - perhaps saying something along the lines of "padding of input images to the network is allowed as long as the non-padding pixels of the input match the size of inputs used by the reference implementation."

**Review by nvpaulius — APPROVED** [review]

File: training_rules.adoc (hunk 1 above)
Comment (petermattson) [reply training_rules.adoc:183]: Fixed to make this clear.
